core/src/main/java/com/klaviyo/coresdk/networking/KlaviyoRequest.kt

```diff
@@ -0,0 +1,22 @@
+package com.klaviyo.coresdk.networking
+
+abstract class KlaviyoRequest: NetworkRequest() {
+    companion object {
+        internal const val BASE_URL = "https://a.klaviyo.com"
+    }
+
+    override var headerData: String? = null
+    override var payload: String? = null
+
+    internal abstract fun buildKlaviyoJsonHeader(): String
+
+    override fun sendNetworkRequest(): String? {
+        headerData = buildKlaviyoJsonHeader()
+        return super.sendNetworkRequest()
+    }
+
+    fun process(): Boolean {
```

Comment:
Function I put in for testing singular network calls for testing purposes (I have a super rudimentary test app I was kicking off requests from and wanted to do some testing before the batching queue is PRed).
I think there's value to keep this here for now